packages/@divvi/mobile/src/i18n/saga.ts

```
@@ -39,13 +41,10 @@ export function* handleFetchOtaTranslations() {
         lastFetchTime !== timestamp ||
         DeviceInfo.getVersion() !== lastFetchAppVersion
       ) {
-        const languageMappings = yield* call([otaClient, otaClient.getLanguageMappings])
-        const customMappedLanguage = _.findKey(languageMappings, { locale: currentLanguage })
-
         const translations = yield* call(
           [otaClient, otaClient.getStringsByLocale],
           undefined,
-          customMappedLanguage || currentLanguage
+          currentLanguage
```

i think as a quick fix for the spanish issue, we can define our own mapping (where "es-419" maps to "es" and continue to use `customMappedLanguage || currentLanguage` here?

---

@kathaypacific but where do I get the mappings from? `getLanguageMappings` doesn't exist anymore.

---

Ah, right. I was trying `es-ES` when I needed to try `es`. Seems to be working. Will adjust the code.

---

thanks! can you please also check the other language codes to see if they still work? we may need to extend the mapping for all languages, i'm not sure what changed under the hood with this upgrade but OTA translations bugs are hard to catch (since there's no visible errors) so i think it's worth checking

---

@kathaypacific so from what I've checked it seems like they [don't expose direct access](https://github.com/crowdin/ota-client-js/blob/30ae36ad0452969d53873b9d15e78ee79a18f25b/src/index.ts) to `language_mapping` or `manifest.json` anymore. The only accessible fields from `manifest.json` are `timestamp` and `content` and the only field that gets the actual full content of the `manifest.json` [is private](https://github.com/crowdin/ota-client-js/blob/30ae36ad0452969d53873b9d15e78ee79a18f25b/src/index.ts#L251C17-L258).

So what I can do is:
1. Whether just copy-paste the content of `language_mapping` from `manifest.json` like this. This is quick but it creates an extra point of maintenance:
```typescript
const customMappedLanguage = _.findKey(
          {
            es: { locale: 'es-419', osx_code: 'es-419.lproj' },
            de: { locale: 'de' },
          },
          { locale: currentLanguage }
        )
```
2. Or I can explicitly fetch manifest file [the same way](https://github.com/crowdin/ota-client-js/blob/30ae36ad0452969d53873b9d15e78ee79a18f25b/src/index.ts#L255) they do in the library and parse `language_mappings`. This is more of an automated approach, although feels more patch-like.

Which one would you prefer?

---

Okay, I've updated the code to try to have the best of both worlds. It will try to fetch the manifest and get data from `language_mapping` if possible, otherwise it will fallback to the hardcoded state of `language_mapping` from `manifest.json` as of now. In this case, we'll at least have a proper current mappings even if their CDNs fail for some reason (not like it [never happened](https://www.thousandeyes.com/blog/cloudflare-outage-analysis-september-16-2024), right? 😄)

---

ah sorry i didn't reply in time. it'd be great not to add a network request that would delay the fetching of the results...i'm also wary that this manifest url may change without any warning since it's not publicly advertised.

it's unlikely that we will be making big changes to our Crowdin setup. so my recommendation is to have a mapping of lang code used in the wallet <> crowdin lang code. e.g. `{ 'es-419': 'es', 'en-US': 'en'}`. it seems to me less brittle than fetching the manifest because we are the ones who define the lang codes, so as long as we don't mess with them in Crowdin, we know the OTA translations should keep working?

---

Sure, I'll set it up as a hardcoded map then. I've noticed there's a bunch of languages that also need new mapping (e.g. Ukrainian is now `uk` instead of `uk-UA`) so I'll also update the mapping.

---

@kathaypacific I've set this up as an extension of `locales` as it seems the most relevant place to store the mappings at. If there's any reason why I should not do it this way (maybe `locales` file should not be imported in this saga for some reason I'm not aware about? idk) – please, let me know. 

I've tested all the locales and all of them seem to work and return all the translations via `getStringsByLocale`.

---

@jeanregisser @kathaypacific I moved it away from locales